pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -998,20 +1006,27 @@ type SemaphoreHolding struct {
 	Holders []string `json:"holders,omitempty" protobuf:"bytes,2,opt,name=holders"`
 }
 
-type WaitingStatus struct {
-	// Holders stores the list of current holder names
-	Holders HolderNames `json:"holders,omitempty" protobuf:"bytes,1,opt,name=holders"`
+type MutexHolding struct {
+	// Name of the mutex
+	Mutex string `json:"mutex,omitempty" protobuf:"bytes,1,opt,name=mutex"`
+	// Holders stores the current mutex holder name.
+	Holder string `json:"holder,omitempty" protobuf:"bytes,2,opt,name=holder"`
 }
 
-type HolderNames struct {
-	// Name stores the name of the resource holding lock
+type MutexStatus struct {
+	// Holding stores the list of resource acquired synchronization lock for workflows.
+	// +listType=atomic
```

Comment:
why atomic?

Reply:
It will just replace the new list during the merge.  Here We just need a new list here